crates/transaction-generator-lib/src/workflow_delegator.rs

```diff
@@ -0,0 +1,239 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    account_generator::AccountGeneratorCreator, accounts_pool_wrapper::AccountsPoolWrapperCreator,
+    call_custom_modules::CustomModulesDelegationGeneratorCreator,
+    entry_points::EntryPointTransactionGenerator, EntryPoints, ObjectPool,
+    ReliableTransactionSubmitter, TransactionGenerator, TransactionGeneratorCreator, WorkflowKind,
+};
+use aptos_logger::{info, sample, sample::SampleRate};
+use aptos_sdk::{
+    transaction_builder::TransactionFactory,
+    types::{transaction::SignedTransaction, LocalAccount},
+};
+use std::{
+    cmp,
+    sync::{
+        atomic::{AtomicUsize, Ordering},
+        Arc,
+    },
+    time::Duration,
+};
+
+/// Wrapper that allows inner transaction generator to have unique accounts
+/// for all transactions (instead of having 5-20 transactions per account, as default)
+/// This is achieved via using accounts from the pool that account creatin can fill,
+/// and burning (removing accounts from the pool) them - basically using them only once.
+/// (we cannot use more as sequence number is not updated on failure)
+struct WorkflowTxnGenerator {
+    stage: StageTracking,
+    generators: Vec<Box<dyn TransactionGenerator>>,
+    pool_per_stage: Vec<Arc<ObjectPool<LocalAccount>>>,
+    num_for_first_stage: usize,
+    completed_for_first_stage: Arc<AtomicUsize>,
```

Comment:
Please add comments for `num_for_first_stage` and `completed_for_first_stage` means.